sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -36,11 +36,12 @@ def logger
 
       # Accepts a resource object that is merged with the default telemetry sdk
       # resource. The use of this method is optional, and is provided as means
-      # to add additional resource information.
+      # to include additional resource information.
+      # If a resource key collision occurs the passed in resource takes priority.
       #
       # @param [Resource] new_resource The resource to be merged
       def resource=(new_resource)
-        @resource = @resource.merge(new_resource)
+        @resource = new_resource.merge(@resource)
```

Comment:
Had the flip it so the supplied resource takes precedence.

Reply:
🤔 is that true? Maybe I need to understand the sequence of initialization...

Reply:
What does the call sequence look like for users? This?
```ruby
OpenTelemetry::SDK.configure do |c|
  c.resource = OpenTelemetry::Resource::Detectors::AutoDetector.detect
  c.resource = OpenTelemetry::SDK::Resources::Resource.create('foo' => 'bar')
end
```

I am always confused by the difference between `Resource#merge` and `Hash#merge` 😞

Reply:
Let me add a test for it so we can see if it's doing the right thing.